nucliadb/src/nucliadb/common/catalog/pg.py

```diff
@@ -26,111 +26,189 @@
 from psycopg import AsyncCursor, sql
 from psycopg.rows import DictRow, dict_row
 
-from nucliadb.common.maindb.pg import PGDriver
+from nucliadb.common.catalog.interface import (
+    Catalog,
+    CatalogExpression,
+    CatalogQuery,
+    CatalogResourceData,
+)
+from nucliadb.common.exceptions import InvalidQueryError
+from nucliadb.common.maindb.driver import Transaction
+from nucliadb.common.maindb.pg import PGDriver, PGTransaction
 from nucliadb.common.maindb.utils import get_driver
-from nucliadb.search.search.query_parser.models import CatalogExpression, CatalogQuery
 from nucliadb_models import search as search_models
-from nucliadb_models.labels import translate_system_to_alias_label
-from nucliadb_models.search import CatalogFacetsRequest, ResourceResult, Resources, SortField, SortOrder
+from nucliadb_models.labels import translate_alias_to_system_label, translate_system_to_alias_label
+from nucliadb_models.search import (
+    CatalogFacetsRequest,
+    ResourceResult,
+    Resources,
+    SortField,
+    SortOrder,
+)
 from nucliadb_telemetry import metrics
 
-from .filters import translate_label
+write_observer = metrics.Observer("pg_catalog_write", labels={"type": ""})
+search_observer = metrics.Observer("pg_catalog_search", labels={"op": ""})
 
-observer = metrics.Observer("pg_catalog_search", labels={"op": ""})
 logger = logging.getLogger(__name__)
 
 SPLIT_REGEX = re.compile(r"\W")
 
 
-def _filter_operands(operands: list[CatalogExpression]) -> tuple[list[str], list[CatalogExpression]]:
-    facets = []
-    nonfacets = []
-    for op in operands:
-        if op.facet:
-            facets.append(op.facet)
-        else:
-            nonfacets.append(op)
+def _pg_transaction(txn: Transaction) -> PGTransaction:
+    return cast(PGTransaction, txn)
 
-    return facets, nonfacets
 
+def pgcatalog_enabled(kbid):
```

Comment:
Maybe this can be deleted? If it's not enabled, you won't even instantiate the PGCatalog class